adobe/spectrum-web-components
# feat(action-bar): Introduction of  custom scoped registry for spectrum web component

## Description
Adding in support for [Scoped Custom Element registries](https://github.com/WICG/webcomponents/blob/gh-pages/proposals/Scoped-Custom-Element-Registries.md) by using the polyfill to enable component libraries to have components defined in the local scope to avoid version mismatch conflicts. 



## Related issue(s)


- https://github.com/adobe/spectrum-web-components/issues/3729


## Motivation and context


This helps in faster development and removes the host dependencies for ace-components. For Implementing it a npm package like [@lit-labs/scoped-registry-mixin](https://www.npmjs.com/package/@lit-labs/scoped-registry-mixin?activeTab=readme) could be used which creates scoped registry for the class & also passes the registry to attachShadow().
cc:: @rickharris 
## How has this been tested?




-   [ ] StoryBook Manual Test :: Scoped Action Bar component should load the  element dependencies in it's local scope 
    1.  Launch the storybook 
    2. Navigate to ActionBarStory and check Scoped stories. 

## Screenshots (if appropriate)


This creates a custom registry for `ActionBarScoped` which also defines the elements it internally uses to the scoped registry you can see that in the screenshot provided below `sp-action-bar-scoped` shadow-root has customElement defined within it scope. This enables us to use multiple version of same element on the same page.
![ActionBar Scoped Registry](https://github.com/adobe/spectrum-web-components/assets/29114607/4dacad50-58bc-4bc9-9612-4f6a4da022f6)


## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
